File: Module/PlayWays/src/main/java-room-grab/com/module/playways/grab/room/view/chorus/ChorusSelfSingCardView.java
package com.module.playways.grab.room.view.chorus;

import android.view.View;
import android.view.ViewStub;

import com.module.playways.R;
import com.module.playways.grab.room.GrabRoomData;
import com.module.playways.grab.room.model.ChorusRoundInfoModel;
import com.module.playways.grab.room.model.GrabRoundInfoModel;
import com.module.playways.grab.room.view.SingCountDownView2;
import com.module.playways.grab.room.view.control.SelfSingCardView;

import java.util.List;


/**
 * 合唱的歌唱者看到的板子
 */
public class ChorusSelfSingCardView extends BaseChorusSelfCardView {

    public final static String TAG = "ChorusSelfSingCardView";

    SingCountDownView2 mSingCountDownView;

    public ChorusSelfSingCardView(ViewStub viewStub, GrabRoomData roomData) {
        super(viewStub, roomData);
    }

    @Override
    protected void init(View parentView) {
        super.init(parentView);
        mSingCountDownView = mParentView.findViewById(R.id.sing_count_down_view);
        mSingCountDownView.setListener(mListener);
    }

    @Override
    protected int layoutDesc() {
        return R.layout.grab_chorus_self_sing_card_stub_layout;
    }

    public boolean playLyric() {
        if(super.playLyric()){
            GrabRoundInfoModel infoModel = mRoomData.getRealRoundInfo();
            mSingCountDownView.startPlay(0, infoModel.getSingTotalMs(), true);
            return true;
        }else{
            return false;
        }
    }

    public void setListener(SelfSingCardView.Listener listener) {
        super.setListener(listener);
    }

    @Override
    public void setVisibility(int visibility) {
        super.setVisibility(visibility);
        if (visibility == View.GONE) {
            if (mSingCountDownView != null) {
                mSingCountDownView.reset();
            }
        }
    }
}

File: Module/PlayWays/src/main/java-room-grab/com/module/playways/grab/room/view/chorus/ChorusSelfSingCardView.kt
package com.module.playways.grab.room.view.chorus

import android.view.View
import android.view.ViewStub
import com.common.core.view.setDebounceViewClickListener
import com.module.playways.R


/**
 * 合唱的歌唱者看到的板子
 */
class ChorusSelfSingCardView(viewStub: ViewStub) : BaseChorusSelfCardView(viewStub) {

//    internal var mSingCountDownView: SingCountDownView2? = null

    override val isForVideo: Boolean
        get() = false

    override fun init(parentView: View) {
        super.init(parentView)
//        mSingCountDownView = mParentView!!.findViewById(R.id.sing_count_down_view)
//        mSingCountDownView?.setListener(mOverListener)
        parentView.findViewById<View>(R.id.relative_container).setDebounceViewClickListener { }
    }

    override fun layoutDesc(): Int {
        return R.layout.grab_chorus_self_sing_card_stub_layout
    }

    public override fun playLyric(): Boolean {
        if (super.playLyric()) {
//            if (H.isGrabRoom()) {
//                mSingCountDownView?.startPlay(0, H.grabRoomData!!.realRoundInfo!!.singTotalMs, true)
//            } else if (H.isMicRoom()) {
//                mSingCountDownView?.startPlay(0, H.micRoomData!!.realRoundInfo!!.singTotalMs, true)
//            }
            return true
        } else {
            return false
        }
    }

    override fun setVisibility(visibility: Int) {
        super.setVisibility(visibility)
//        if (visibility == View.GONE) {
//            mSingCountDownView?.reset()
//        }
    }
}
